Bottom: 24163c229111bc35e7f240157d3d5ae2a8c86fc5
Top:    41f331767919f0ba078cb7d78b5407fadbbb2da8
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-09-25 11:49:34 +0800

Refresh of introduce-core-ring

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index ea65e04..8f9f6ad 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -3037,7 +3037,7 @@ struct skb_array *tun_get_skb_array(struct file *file)
 }
 EXPORT_SYMBOL_GPL(tun_get_skb_array);
 
-struct skb_array *tun_get_xdp_ring(struct file *file)
+struct core_ring *tun_get_xdp_ring(struct file *file)
 {
 	struct tun_file *tfile;
 
@@ -3046,9 +3046,9 @@ struct skb_array *tun_get_xdp_ring(struct file *file)
 	tfile = file->private_data;
 	if (!tfile)
 		return ERR_PTR(-EBADFD);
-	return &tfile->xdp_ring;
+	return &tfile->xdp_ring->ring;
 }
-EXPORT_SYMBOL_GPL(tun_get_skb_array);
+EXPORT_SYMBOL_GPL(tun_get_xdp_ring);
 
 module_init(tun_init);
 module_exit(tun_cleanup);
